File: es_07.cpp
// 7. Inserire da tastiera finché non inserisco -1,
// calcolare la media dei numeri inseriti e
// la percentuale dei numeri pari.

#include <iostream>
using namespace std;

int main() {
    int n, somma = 0, conta_pari = 0, conta_numeri = 0;
    float media, percentuale_pari;

    do {
        cout << "Inserisci un numero: ";
        cin >> n;

        if (n != -1) {
            somma += n;
            conta_numeri++;

            if (n % 2 == 0) {
                conta_pari++;
            }
        }
    } while (n != -1);

    if (conta_numeri > 0) {
        media = float(somma) / conta_numeri;
        percentuale_pari = (float(conta_pari) / conta_numeri) * 100;

        cout << "La somma è: " << somma << endl
             << "La media è: " << media << endl
             << "Percentuale di numeri pari: " << percentuale_pari << "%" << endl;
    } else {
        cout << "Nessun numero inserito." << endl;
    }

    return 0;
}

File: Python/es_07.py
# 7. Inserire da tastiera finché non inserisco -1,
# calcolare la media dei numeri inseriti e
# la percentuale dei numeri pari.

i = somma = n_pari = 0

while True:
    n = int(
        input('Inserisci un numero: ')
    )

    if n == -1: break

    somma += n
    i += 1

    if n % 2 == 0:
        n_pari += 1

if i > 0:
    media = somma / i
    perc_pari = (n_pari / i) * 100

    print(
        f'Media dei numeri inseriti: {media}',
        f'Percentuale dei numeri pari: {perc_pari}%', sep='\n'
    )
else:
    print('Nessun numero inserito.')
